File: src/main/webapp/default-site/static-assets/components/cstudio-admin/mods/audit-log.js
/*
 * Crafter Studio Web-content authoring solution
 * Copyright (C) 2007-2016 Crafter Software Corporation.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

CStudioAuthoring.Utils.addCss("/static-assets/components/cstudio-admin/mods/audit-log.css");
CStudioAdminConsole.Tool.AuditLog = CStudioAdminConsole.Tool.AuditLog ||  function(config, el)  {
	this.containerEl = el;
	this.config = config;
	this.types = [];
	return this;
};
var list = [];
var auditLog = [];
/**
 * Overarching class that drives the content type tools
 */
YAHOO.extend(CStudioAdminConsole.Tool.AuditLog, CStudioAdminConsole.Tool, {
	renderWorkarea: function() {
		var workareaEl = document.getElementById("cstudio-admin-console-workarea");
		
		workareaEl.innerHTML = 
			"<div id='auditlog-list'>" +
			"</div>";
			
			var actions = [];

			CStudioAuthoring.ContextualNav.AdminConsoleNav.initActions(actions);
			
			this.renderAuditLogList();
	},
	
	renderAuditLogList: function() {
			
		this.renderAuditLogTable();

	},
	
	renderAuditLogTable: function () {
		var auditLogListEl = document.getElementById("auditlog-list");
		auditLogListEl.innerHTML =
		"<table id='auditLogTable' class='cs-auditloglist'>" +
			 	"<tr>" +
				 	"<th class='cs-auditloglist-heading'>"+CMgs.format(langBundle, "setAuditLogTabID")+"</th>" +
    			 	"<th class='cs-statelist-heading'>"+CMgs.format(langBundle, "setAuditLogTabUser")+"</th>" +
                    "<th class='cs-statelist-heading'>"+CMgs.format(langBundle, "setAuditLogTabPath")+"</th>" +
                    "<th class='cs-statelist-heading'>"+CMgs.format(langBundle, "setAuditLogTabContentType")+"</th>" +
                    "<th class='cs-statelist-heading'>"+CMgs.format(langBundle, "setAuditLogTabModifiedDate")+"</th>" +
				 	"<th class='cs-statelist-heading'>"+CMgs.format(langBundle, "setAuditLogTabActivityType")+"</th>" +
				 "</tr>" + 
			"</table>";
	
			cb = {
				success: function(response) {
					var aLog = eval("(" + response.responseText + ")");
					auditLog = aLog.items;
					
					var auditLogTableEl = document.getElementById("auditLogTable");
					for(var i=0; i<aLog.items.length; i++) {
						var logItem = aLog.items[i];
						var trEl = document.createElement("tr");
						     
						var rowHTML =
				 			"<td class='cs-auditloglist-detail-id'>" + logItem.id + "</td>" +
				 			"<td class='cs-auditloglist-detail'>" + logItem.userId + "</td>" +
                            "<td class='cs-auditloglist-detail'>" + logItem.contentId + "</td>" +
                            "<td class='cs-auditloglist-detail'>" + logItem.contentType + "</td>" +
                            "<td class='cs-auditloglist-detail'>" + logItem.modifiedDate + "</td>" +
                            "<td class='cs-auditloglist-detail'>" + logItem.type + "</td>";
				 		trEl.innerHTML = rowHTML;
				 		auditLogTableEl.appendChild(trEl);
					}
				},
				failure: function(response) {
				},
				self: this
			};
			
			var serviceUri = "/api/1/services/api/1/activity/get-audit-log.json?site="+CStudioAuthoringContext.site+"&startpos=0&num=20";

			YConnect.asyncRequest("GET", CStudioAuthoring.Service.createServiceUri(serviceUri), cb);
	}

});

CStudioAuthoring.Module.moduleLoaded("cstudio-console-tools-audit-log",CStudioAdminConsole.Tool.AuditLog);

File: src/main/java/org/craftercms/studio/api/v2/dal/AuditLog.java
/*
 * Copyright (C) 2007-2022 Crafter Software Corporation. All Rights Reserved.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License version 3 as published by
 * the Free Software Foundation.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package org.craftercms.studio.api.v2.dal;

import java.time.ZonedDateTime;
import java.util.List;

public class AuditLog {
    private long id;
    private long organizationId;
    private long siteId;
    private String siteName;
    private String operation;
    private ZonedDateTime operationTimestamp;
    private String origin;
    private String primaryTargetId;
    private String primaryTargetType;
    private String primaryTargetSubtype;
    private String primaryTargetValue;
    private String actorId;
    private String actorDetails;
    private String clusterNodeId;
    private List<AuditLogParameter> parameters;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public long getOrganizationId() {
        return organizationId;
    }

    public void setOrganizationId(long organizationId) {
        this.organizationId = organizationId;
    }

    public long getSiteId() {
        return siteId;
    }

    public void setSiteId(long siteId) {
        this.siteId = siteId;
    }

    public String getSiteName() {
        return siteName;
    }

    public void setSiteName(String siteName) {
        this.siteName = siteName;
    }

    public String getOperation() {
        return operation;
    }

    public void setOperation(String operation) {
        this.operation = operation;
    }

    public ZonedDateTime getOperationTimestamp() {
        return operationTimestamp;
    }

    public void setOperationTimestamp(ZonedDateTime operationTimestamp) {
        this.operationTimestamp = operationTimestamp;
    }

    public String getOrigin() {
        return origin;
    }

    public void setOrigin(String origin) {
        this.origin = origin;
    }

    public String getPrimaryTargetId() {
        return primaryTargetId;
    }

    public void setPrimaryTargetId(String primaryTargetId) {
        this.primaryTargetId = primaryTargetId;
    }

    public String getPrimaryTargetType() {
        return primaryTargetType;
    }

    public void setPrimaryTargetType(String primaryTargetType) {
        this.primaryTargetType = primaryTargetType;
    }

    public String getPrimaryTargetSubtype() {
        return primaryTargetSubtype;
    }

    public void setPrimaryTargetSubtype(String primaryTargetSubtype) {
        this.primaryTargetSubtype = primaryTargetSubtype;
    }

    public String getPrimaryTargetValue() {
        return primaryTargetValue;
    }

    public void setPrimaryTargetValue(String primaryTargetValue) {
        this.primaryTargetValue = primaryTargetValue;
    }

    public String getActorId() {
        return actorId;
    }

    public void setActorId(String actorId) {
        this.actorId = actorId;
    }

    public String getActorDetails() {
        return actorDetails;
    }

    public void setActorDetails(String actorDetails) {
        this.actorDetails = actorDetails;
    }

    public String getClusterNodeId() {
        return clusterNodeId;
    }

    public void setClusterNodeId(String clusterNodeId) {
        this.clusterNodeId = clusterNodeId;
    }

    public List<AuditLogParameter> getParameters() {
        return parameters;
    }

    public void setParameters(List<AuditLogParameter> parameters) {
        this.parameters = parameters;
    }
}
